chain-factory/src/factory.rs

```diff
@@ -1,7 +1,25 @@
 use chain_config::StakeMultiArg;
+use multiversx_sc::codec;
+use multiversx_sc::derive::{ManagedVecItem, TypeAbi};
+use multiversx_sc::proxy_imports::TopDecode;
 
 multiversx_sc::imports!();
 
+#[derive(TopDecode, TypeAbi, ManagedVecItem, Clone)]
+pub struct ContractMap<M: ManagedTypeApi> {
+    pub name: ManagedBuffer<M>,
+    pub address: ManagedAddress<M>,
+}
+
+const SC_ARRAY: [&str; 6] = [
```

Comment:
Might want to expose these to a view so they don't have to scavenge through code to find them.

Also, for efficiency, use `&[u8]` instead of `&str", and add a `b` in front of the strings.

Reply:
but how can I check if the contracts name are the correct ones if I put them inside a view ?